geopandas/vectorized.pyx

```diff
@@ -951,34 +908,32 @@ cdef get_coordinate_point(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms,
                           int coordinate):
     """ Get x, y, or z value for an array of points """
     cdef Py_ssize_t idx
-    cdef GEOSContextHandle_t handle
     cdef GEOSGeometry *geom
     cdef GEOSCoordSequence *sequence
     cdef unsigned int n = geoms.size
     cdef double value
 
     cdef np.ndarray[double, ndim=1, cast=True] out = np.empty(n, dtype=np.float)
 
-    handle = get_geos_context_handle()
-
-    if coordinate == 0:
-        func = GEOSCoordSeq_getX_r
-    elif coordinate == 1:
-        func = GEOSCoordSeq_getY_r
-    elif coordinate == 2:
-        func = GEOSCoordSeq_getZ_r
-    else:
-        raise NotImplementedError("Coordinate must be between 0-x, 1-y, 2-z")
-
-    with nogil:
-        for idx in xrange(n):
-            geom = <GEOSGeometry *> geoms[idx]
-            if geom is not NULL:
-                sequence = GEOSGeom_getCoordSeq_r(handle, geom)
-                func(handle, sequence, 0, &value)
-                out[idx] = value
-            else:
-                out[idx] = 0
+    with get_handle() as handle:
+        if coordinate == 0:
+            func = GEOSCoordSeq_getX_r
+        elif coordinate == 1:
+            func = GEOSCoordSeq_getY_r
+        elif coordinate == 2:
+            func = GEOSCoordSeq_getZ_r
+        else:
+            raise NotImplementedError("Coordinate must be between 0-x, 1-y, 2-z")
```

Comment:
Does this needs to be inside the handle? (in the other cases I thought it was outside the context)

Reply:
Good catch.  Fixed.